xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-api/src/main/java/org/xwiki/ratings/internal/averagerating/XObjectAverageRatingManager.java

```diff
@@ -115,8 +124,84 @@ public long removeAverageRatings(EntityReference entityReference) throws Ratings
     public long moveAverageRatings(EntityReference oldReference, EntityReference newReference)
         throws RatingsException
     {
-        // We don't need to do anything here: if a document reference has been moved then the xobject move with it.
-        return 0;
+        // Checks that we are in the presence of entity types that can be handled by the move method.
+        if (oldReference == null
+            || newReference == null
+            || oldReference.extractReference(DOCUMENT) == null
+            || newReference.extractReference(DOCUMENT) == null)
+        {
+            throw new RatingsException(
+                "Impossible to move the average ratings from [" + oldReference + "] to [" + newReference + "].");
+        }
+
+        // The list of the average ratings updated during the move.
+        List<AverageRating> changedAverageRatings = new ArrayList<>();
+
+        try {
+            XWikiDocument actualDoc = (XWikiDocument) this.documentAccessBridge.getDocumentInstance(newReference);
+
+            // We must inspect all the XObject to see of they need to be updated, some of them can be attached to
+            // sub-elements of the page.
+            for (BaseObject ratingsObject : actualDoc
+                .getXObjects(AverageRatingClassDocumentInitializer.AVERAGE_RATINGS_CLASSREFERENCE)) {
+                moveAverageRatingsObject(oldReference, newReference, ratingsObject, changedAverageRatings);
+            }
+
+            // Save the document if something has changed
+            if (!changedAverageRatings.isEmpty()) {
+                XWikiContext context = this.contextProvider.get();
+                this.getObservationManager()
+                    .notify(new UpdatingAverageRatingEvent(), this.getIdentifier(), changedAverageRatings);
+                try {
+                    context.getWiki().saveDocument(actualDoc, "Move average ratings XObjects", true, context);
+                    this.getObservationManager()
+                        .notify(new UpdatedAverageRatingEvent(), this.getIdentifier(), changedAverageRatings);
+                } catch (XWikiException e) {
+                    this.getObservationManager()
+                        .notify(new UpdateAverageRatingFailedEvent(), this.getIdentifier(), changedAverageRatings);
+                }
+            }
+        } catch (Exception e) {
+            throw new RatingsException(
+                String.format("Error while moving the average ratings from [%s] to [%s]", oldReference, newReference),
+                e);
+        }
+
+        return changedAverageRatings.size();
+    }
+
+    private void moveAverageRatingsObject(EntityReference oldDocumentEntityReference,
+        EntityReference newDocumentEntityReference,
+        BaseObject ratingsObject, List<AverageRating> changedAverageRatings)
```

Comment:
Still not a big fan of having a method not returning anything and filling a list. Here it would be easy to return an xobject and to fill the list in the main method.